src/main/java/jenkins/plugins/slack/pipeline/SlackUserIdsFromCommittersStep.java

```diff
@@ -0,0 +1,164 @@
+        }
+
+        @Nonnull
+        @Override
+        public String getDisplayName() {
+            return Messages.slackUserIdsFromCommittersDisplayName();
+        }
+
+        public ListBoxModel doFillTokenCredentialIdItems(@AncestorInPath final Item item) {
+            return findTokenCredentialIdItems(item);
+        }
+    }
+
+    public static class SlackUserIdsFromCommittersStepExcecution extends SynchronousNonBlockingStepExecution<List<String>> {
+
+        private static final long serialVersionUID = 1L;
+
+        private transient final SlackUserIdsFromCommittersStep step;
+
+        SlackUserIdsFromCommittersStepExcecution(final SlackUserIdsFromCommittersStep step, final StepContext context) {
+            super(context);
+            this.step = step;
+        }
+
+        @Override
+        protected List<String> run() throws Exception {
+
+            final Jenkins jenkins = Jenkins.get();
+            final Item item = getItemForCredentials(getContext());
+            final SlackNotifier.DescriptorImpl slackDesc = jenkins
+                    .getDescriptorByType(SlackNotifier.DescriptorImpl.class);
+
+            final String tokenCredentialId = step.tokenCredentialId != null ? step.tokenCredentialId
+                    : slackDesc.getTokenCredentialId();
+            final boolean botUser = step.botUser || slackDesc.isBotUser();
+            final SlackUserIdResolver userIdResolver = slackDesc.getSlackUserIdResolver();
+
+            final Run run = getContext().get(Run.class);
+            final TaskListener listener = getContext().get(TaskListener.class);
+            Objects.requireNonNull(listener, "Listener is mandatory here");
+
+            if (!botUser) {
+                listener.getLogger().println("The slackUserIdFromEmail step requires bot user mode");
+                return null;
+            }
+
+            final String populatedToken;
+            try {
+                populatedToken = CredentialsObtainer.getTokenToUse(tokenCredentialId, item, null);
+            } catch (final IllegalArgumentException e) {
+                listener.error(Messages.notificationFailedWithException(e));
+                return null;
+            }
+
+            final List<String> slackUserIds = new ArrayList<>();
+            try (CloseableHttpClient client = getHttpClient()) {
+                // include committer userIds in roomIds
+                if (botUser && userIdResolver != null && run != null) {
+                    userIdResolver.setAuthToken(populatedToken);
+                    userIdResolver.setHttpClient(client);
+                    final List<String> userIds = userIdResolver.resolveUserIdsForRun(run);
+                    slackUserIds
+                            .addAll(userIds.stream().filter(Objects::nonNull).distinct().collect(Collectors.toList()));
+                }
+            } catch (final IOException e) {
+                logger.log(Level.WARNING, "Error closing HttpClient", e);
```

Comment:
Is this the only reason the IOException could occur?

Reply:
Yes, as far as I can tell because none of the called methods throw exceptions.